internal/users/locking/locking_bwrap_test.go

```diff
@@ -0,0 +1,358 @@
+	require.NotEmpty(t, testLockerUtility, "Setup: Locker utility unset")
+
+	groupFile := filepath.Join("/etc", "group")
+	groupContents := "testgroup:x:1001:testuser"
+
+	//nolint:gosec // G306 The group file is expected to have permissions 0644
+	err := os.WriteFile(groupFile, []byte(groupContents), 0644)
+	require.NoError(t, err, "Writing group file")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cmd := exec.CommandContext(ctx, testLockerUtility)
+	t.Logf("Running command: %s", cmd.Args)
+
+	lockerExited := make(chan error)
+	writeLockExited := make(chan error)
+	t.Cleanup(func() {
+		cancel()
+		syscall.Kill(cmd.Process.Pid, syscall.SIGKILL)
+		require.Error(t, <-lockerExited, "Stopping locking process")
+		require.NoError(t, <-writeLockExited, "Final locking")
+		require.NoError(t, userslocking.WriteUnlock(), "Final unlocking")
+	})
+
+	go func() {
+		lockerExited <- cmd.Run()
+	}()
+
+	select {
+	case <-time.After(sleepDuration(1 * time.Second)):
+		t.Cleanup(func() { cmd.Process.Kill() })
+		// If we're time-outing: it's fine, it means the test-locker process is running
+	case err := <-lockerExited:
+		require.NoError(t, err, "test locker should have not failed")
+	}
+
+	gPasswdExited := make(chan error)
+	go func() {
+		_, err := runGPasswd(t, "--add", "root", "testgroup")
+		gPasswdExited <- err
+	}()
+
+	select {
+	case <-time.After(sleepDuration(3 * time.Second)):
+		// If we're time-outing: it's fine, it means we were locked!
+	case err := <-gPasswdExited:
+		require.ErrorIs(t, err, userslocking.ErrLock, "GPasswd should fail")
+	}
+
+	go func() {
+		writeLockExited <- userslocking.WriteLock()
+	}()
+
+	select {
+	case <-time.After(sleepDuration(3 * time.Second)):
+		// If we're time-outing: it's fine, it means the test-locker process is
+		// still running and holding the lock.
+	case err := <-writeLockExited:
+		require.ErrorIs(t, err, userslocking.ErrLock, "Locking should not work")
+	}
+}
+
+func TestLockingLockedDatabaseFailsAfterTimeout(t *testing.T) {
+	require.Zero(t, os.Geteuid(), "Not root")
+
+	testLockerUtility := os.Getenv("AUTHD_TESTS_PASSWD_LOCKER_UTILITY")
+	require.NotEmpty(t, testLockerUtility, "Setup: Locker utility unset")
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cmd := exec.CommandContext(ctx, testLockerUtility)
+	t.Logf("Running command: %s", cmd.Args)
+
+	lockerExited := make(chan error)
+	t.Cleanup(func() {
+		cancel()
+		syscall.Kill(cmd.Process.Pid, syscall.SIGKILL)
+		require.Error(t, <-lockerExited, "Stopping locking process")
+		require.NoError(t, userslocking.WriteUnlock(), "Final unlocking")
+	})
+
+	go func() {
+		lockerExited <- cmd.Run()
+	}()
+
+	select {
+	case <-time.After(sleepDuration(1 * time.Second)):
+		t.Cleanup(func() { cmd.Process.Kill() })
+		// If we're time-outing: it's fine, it means the test-locker process is running
+	case err := <-lockerExited:
+		require.NoError(t, err, "test locker should have not failed")
+	}
```

Comment:
Instead of waiting for a second to check if the test locker fails with an unexpected error, we can make `WriteLock` return an `ErrTimeout` error which wraps an `ErrLock`, which we can then check for below:

In `locking.go`:
```
// ErrTimeout is the error when locking the database times out.
ErrTimeout = fmt.Errorf("%w: timeout", ErrLock)
```
```
	case <-time.After(16 * time.Second):
		return ErrTimeout
```

In `TestLockingLockedDatabaseFailsAfterTimeout`:
```
err := <-writeLockExited
t.Log("Done waiting for lock!")
require.ErrorIs(t, err, userslocking.ErrTimeout)
```